nautobot/extras/models/relationships.py

```diff
@@ -233,6 +235,16 @@ def get_for_model(self, model):
             self.get_queryset().filter(destination_type=content_type),
         )
 
+    def get_required_for_model(self, model):
+        """
+        Return a queryset with all required Relationships on the given model.
+        """
+        content_type = ContentType.objects.get_for_model(model._meta.concrete_model)
+        return self.get_queryset().filter(
+            (Q(source_type=content_type) | Q(destination_type=content_type))
+            & Q(required_side__in=["source", "destination"])
+        )
```

Comment:
This "feels" wrong to me. As implemented, won't it return cases where this model is on the *non*-required side of a required relationship as well?